src/main.rs

```diff
@@ -85,6 +87,15 @@ fn setup_logger(log_target: LogTarget, verbose: bool) -> eyre::Result<()> {
 }
 
 fn main() -> eyre::Result<()> {
+    // Start with superset of all potentially required promises.
+    // Drop later after CLI arguments and configuration files were parsed.
+    #[cfg(target_os = "openbsd")]
+    pledge(
+        "stdio rpath wpath cpath inet mcast flock chown unix dns proc exec audio",
+        None,
+    )
+    .unwrap();
```

Comment:
I'm not sure, what the scenarios are, where this fails, but from a user's perspective it might be good to include a short explanation what failed here? Like "failed to restrict service permissions via pledge" or something similar?

Reply:
https://man.openbsd.org/pledge.2#ERRORS
```
pledge() will fail if:

EFAULT    promises or execpromises points outside the process's allocated address space.
EINVAL    promises is malformed or contains invalid keywords.
EPERM     This process is attempting to increase permissions.
```
Promises are literal values, so the first two won't happen, leaving just `EPERM`.

In any event, first thing to do (upon pledge(2) failure) on OpenBSD would be to run the program through [ktrace(1)](https://man.openbsd.org/ktrace.1) which gathers all relevant information.

`unwrap()` seems enough, is used in the pledge crate's own example and somewhat matches the usual C idiom:
```
if (pledge(..., NULL) == -1)
    err(1, "pledge");
```

Reply:
That sounds very good. Thank you for the clarification.